.github/workflows/apk-builder.yml

```diff
@@ -0,0 +1,89 @@
+name: APK build
+on:
+  workflow_dispatch:
+
+permissions:
+    contents: write
+      
+jobs:
+    build:
+      name: Build Release APK
+      runs-on: ubuntu-latest
+  
+      steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+  
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      # Load the google-services.json file from the secrets
+      - name: Load Google Service file
+        env:
+          DATA: ${{ secrets.GOOGLE_SERVICES_JSON }}
+        run: echo $DATA | base64 -di > app/google-services.json
+
+      # Load the keystore file from the secrets
+      - name: Load Keystore file
+        env:
+          DATA: ${{ secrets.KEYSTORE_JKS}}
+        run: echo $DATA | base64 -di > app/upload-keystore.jks
+
+      # Load the keystore.properties file from the secrets
+      - name: Load Keystore properties file
+        env:
+          DATA: ${{ secrets.KEYSTORE}}
+        run: echo $DATA | base64 -di > keystore.properties
+      
+      # Load the secrets
+      - name: Load secrets
+        env:
+          DATA: ${{ secrets.SECRETS }}
+        run: |
+          echo $DATA | base64 -di > secrets.properties
+          echo $DATA | base64 -di > local.defaults.properties
+  
+      - name: Grant execute permission for gradlew
+        run: chmod +x ./gradlew
+  
+      - name: Build APK
+        run: ./gradlew assembleRelease
+
+      - name: Generate release tag
+        id: tag
+        run: |
+          echo "release_tag=streetworkapp_$(date +"%Y.%m.%d_%H-%M")" >> $GITHUB_OUTPUT
+      
+      - name: Rename APK
+        run: mv app/build/outputs/apk/release/app-release.apk app/build/outputs/apk/release/${{ steps.tag.outputs.release_tag }}.apk
+
+      - name: Upload APK in artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: streetworkapp
+          path: app/build/outputs/apk/release/${{ steps.tag.outputs.release_tag }}.apk
+          if-no-files-found: error
+          retention-days: 90
+
+      - name: Load Google Maps API Key
+        env:
+          MAPS_API_KEY: ${{ secrets.MAPS_API_KEY }}
+        run: echo "MAPS_API_KEY=${MAPS_API_KEY}" >> local.defaults.properties
```

Comment:
fix: the google maps API key is currently stored in the LOCAL_PROPERTIES secret, furthermore it is stored in base 64 so you might have to decode it before using it

Reply:
Yep done ✅